openml_OS/models/Estimation_procedure.php

```diff
@@ -37,61 +37,76 @@ function get_by_parameters( $ttid, $type, $repeats, $folds, $percentage, $strati
     $percentage = ' AND percentage ' . (($percentage == NULL) ? ' IS NULL ' : ' = ' . $percentage);
     $stratified = ' AND stratified_sampling ' . (($stratified == NULL) ? ' IS NULL ' : ' = ' . $stratified);
     
-    $ep = $this->getWhere( $task_type . $type . $repeats . $folds . $percentage );
-    if($ep === false) 
+    $ep = $this->getWhere($task_type . $type . $repeats . $folds . $percentage);
+    if ($ep === false) {
       return false;
-    else
+    } else {
       return end($ep);
+    }
   }
   
-  function splits_arff_size( $ep, $instances ) {
-    if( $ep->type == 'crossvalidation' ) {
-      return $instances * $ep->folds * $ep->repeats;
-    } elseif( $ep->type == 'leaveoneout' ) {
-      return $instances * $instances;
-    } elseif( $ep->type == 'holdout' ) {
-      return $instances * $ep->repeats;
-    } elseif( $ep->type == 'testthentrain' ) {
-      return 0;
-    } elseif( $ep->type == 'learningcurve' ) {
-      $foldsize = ceil($instances / $ep->folds);
-      $trainingsetsize = $foldsize * ($ep->folds-1);
-      $totalsize = 0;
-      for( $i = 0; $i < $this->number_of_samples( $trainingsetsize ); ++$i ) {
-        // size of the sample and size of the 'test set'
-        $totalsize += $this->sample_size($i, $trainingsetsize ) + $foldsize;
-      }
-      return $totalsize * $ep->folds * $ep->repeats; 
+  public function trainingset_size($datasetsize, $ep_procedure) {
+    if ($ep_procedure->type == 'crossvalidation') {
+      $folds = $ep_procedure->folds;
+      $foldsize = ceil($datasetsize / $folds);
+      return $foldsize * ($folds-1);
+    } elseif ($ep_procedure->type == 'leaveoneout') {
+      return $datasetsize - 1;
     } else {
-      // TODO: implement other types.
-      return -1;
+      throw new Exception('Trainset size procedure not determined yet');
     }
   }
   
-  public function trainingset_size( $datasetsize, $folds ) {
-    $foldsize = ceil($datasetsize / $folds);
-    return $foldsize * ($folds-1);
-  }
-  
-  public function number_of_samples( $trainingsetsize ) {
+  public function number_of_samples($trainingsetsize) {
     $i = 0;
-    for( ; $this->sample_size( $i, $trainingsetsize ) < $trainingsetsize; ++$i ) { }
+    for( ; $this->sample_size($i, $trainingsetsize) < $trainingsetsize; ++$i ) { }
```

Comment:
Weird construct. Why is $i defined outside of the loop?

Reply:
Looks like we want `$i` to have a broader scope. Was already the case, this line was not really changed.